stdlib/public/core/Integers.swift

```diff
@@ -2755,6 +2770,16 @@ extension FixedWidthInteger {
     lhs = _nonMaskingRightShift(lhs, shift)
   }
 
+  @inlinable
+  public func isMultiple(of other: Self) -> Bool {
+    // Nothing but zero is a multiple of zero.
+    if other == 0 { return self == 0 }
+    // Special case to avoid overflow on .min / -1 for signed types.
+    if Self.isSigned && self == .min && other == -1 { return true }
```

Comment:
I'd be curious if performance changes if the two conditionals were replaced with a check of the magnitude, i.e. `return other.magnitude <= 1 || self % other == 0`...

Reply:
For concrete types, there should basically be no difference in the face of optimization. We can actually re-arrange this slightly to have only a single test on the fast path, but the first thing is to get the API in.